src/MySqlConnector/Serialization/MySqlSession.cs

```diff
@@ -526,47 +526,81 @@ private void VerifyConnected()
 
 		private async Task InitSslAsync(ProtocolCapabilities serverCapabilities, ConnectionSettings cs, IOBehavior ioBehavior, CancellationToken cancellationToken)
 		{
-			X509CertificateCollection clientCertificates = null;
+			X509CertificateCollection CertificateFiles = null;
 			if (cs.CertificateFile != null)
 			{
 				try
 				{
-					var certificate = new X509Certificate2(cs.CertificateFile, cs.CertificatePassword);
-					clientCertificates = new X509CertificateCollection {certificate};
+					X509Certificate2 certificate;
+					if (cs.CertificatePassword == null)
+						certificate = new X509Certificate2(cs.CertificateFile);
+					else
+						certificate = new X509Certificate2(cs.CertificateFile, cs.CertificatePassword);
```

Comment:
Is this `if`/`else` necessary? It looks like the old code worked to pass a potentially-null `cs.CertificatePassword` as the second argument to the constructor, and I don't see any indication in the documentation that it can't be `null`. In corefx, the [single-argument constructor](https://github.com/dotnet/corefx/blob/master/src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509Certificate.cs#L110) just seems to pass `null` when it delegates to an overload.

So it feels like this could be simplified back to the old code.